src/customFileParser/unpackGeometry/InstancedMesh.ts

```diff
@@ -0,0 +1,89 @@
+import * as THREE from 'three';
+import { InstancedMeshGroup, InstancedMesh, InstancedMeshCollection } from '../../geometry/InstancedMeshGroup';
+import { CompressedGeometryData, UncompressedValues } from './../sharedFileParserTypes';
+import PropertyLoader from './../PropertyLoader';
+import { xAxis, yAxis, zAxis } from './../../constants';
+import SceneStats from './../../SceneStats';
+import Sector from './../../Sector';
+
+const matrix = new THREE.Matrix4();
+const rotation = new THREE.Matrix4();
+
+export default function unpackInstancedMeshes(
+  rootSector: Sector,
+  uncompressedValues: UncompressedValues,
+  sectorPathToInstancedMeshData: {[path: string]: CompressedGeometryData},
+  sceneStats: SceneStats,
+  treeIndexNodeIdMap: number[],
+  colorMap: THREE.Color[]) {
+
+  const data = new PropertyLoader(uncompressedValues);
+  const meshCounts: any = {};
+  const fileIdToSector: {[fileId: string]: Sector} = {};
+
+  // Count meshes per file Id and triangle offset
+  for (const sector of rootSector.traverseSectors()) {
+    const geometryInfo = sectorPathToInstancedMeshData[sector.path];
+    if (geometryInfo !== undefined) {
+      for (let i = 0; i < geometryInfo.count; i++) {
+        data.loadData(geometryInfo);
+        fileIdToSector[data.fileId] = fileIdToSector[data.fileId] ? fileIdToSector[data.fileId] : sector;
+        while (sector.path.indexOf(fileIdToSector[data.fileId]!.path) === -1) {
+          fileIdToSector[data.fileId] = fileIdToSector[data.fileId]!.parent!;
+        }
+        meshCounts[data.fileId] = meshCounts[data.fileId] ? meshCounts[data.fileId] : {};
+        meshCounts[data.fileId][data.triangleOffset] = meshCounts[data.fileId][data.triangleOffset] ?
+          meshCounts[data.fileId][data.triangleOffset] : { count: 0, triangleCount: data.triangleCount };
+        meshCounts[data.fileId][data.triangleOffset].count++;
+      }
+      geometryInfo.indexes.rewind();
+      geometryInfo.nodeIds.rewind();
+    }
+  }
+
+  // Create mesh collections for each file Id and triangle offset
+  const collections: any = {};
+  Object.keys(meshCounts).forEach(fileId => {
+    collections[fileId] = collections[fileId] ? collections[fileId] : {};
+    Object.keys(meshCounts[fileId]).forEach(triangleOffset => {
+      const { count, triangleCount } = meshCounts[fileId][triangleOffset];
+      collections[fileId][triangleOffset] = new InstancedMeshCollection(
+        parseInt(triangleOffset, 10), triangleCount, count);
```

Comment:
Might be obvious, but why do we have to parse int here? `triangleOffset` should already be a number from the beginning shouldn't it?